File: nan21.dnet.module.bp/nan21.dnet.module.bp.ui.extjs/src/main/webapp/src/ds/BusinessPartnerDs.js
 
   	 
Ext.define("net.nan21.dnet.module.bp.md.ds.model.BusinessPartnerDs" , {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"name",type:"string" },
		{name:"code",type:"string" },
		{name:"active",type:"boolean" },
		{name:"notes",type:"string" },
		{name:"id",type:"int", useNull:true },
		{name:"clientId",type:"int", useNull:true },
		{name:"createdAt",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"modifiedAt",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"createdBy",type:"string" },
		{name:"modifiedBy",type:"string" },
		{name:"version",type:"int", useNull:true },
		{name:"type",type:"string" },
		{name:"businessObject",type:"string" },
		{name:"countryId",type:"int", useNull:true },
		{name:"countryCode",type:"string" },
		{name:"taxPayerNo",type:"string" },
		{name:"firstName",type:"string" },
		{name:"lastName",type:"string" },
		{name:"middleName",type:"string" },
		{name:"gender",type:"string" },
		{name:"identityCardNo",type:"string" },
		{name:"passportNo",type:"string" },
		{name:"birthDate",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"companyName",type:"string" },
		{name:"legalFormId",type:"int", useNull:true },
		{name:"legalForm",type:"string" },
		{name:"registrationNo",type:"string" },
		{name:"registrationDate",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"className",type:"string" }
	]
});

Ext.define("net.nan21.dnet.module.bp.md.ds.model.BusinessPartnerDsFilter" , {
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"name",type:"string" },
		{name:"code",type:"string" },
		{name:"active",type:"boolean", useNull:true },
		{name:"notes",type:"string" },
		{name:"id",type:"int", useNull:true },
		{name:"clientId",type:"int", useNull:true },
		{name:"createdAt",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"modifiedAt",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"createdBy",type:"string" },
		{name:"modifiedBy",type:"string" },
		{name:"version",type:"int", useNull:true },
		{name:"type",type:"string" },
		{name:"businessObject",type:"string" },
		{name:"countryId",type:"int", useNull:true },
		{name:"countryCode",type:"string" },
		{name:"taxPayerNo",type:"string" },
		{name:"firstName",type:"string" },
		{name:"lastName",type:"string" },
		{name:"middleName",type:"string" },
		{name:"gender",type:"string" },
		{name:"identityCardNo",type:"string" },
		{name:"passportNo",type:"string" },
		{name:"birthDate",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"companyName",type:"string" },
		{name:"legalFormId",type:"int", useNull:true },
		{name:"legalForm",type:"string" },
		{name:"registrationNo",type:"string" },
		{name:"registrationDate",type:"date" ,dateFormat:Ext.MODEL_DATE_FORMAT },
		{name:"className",type:"string" }
	] 
});
	
Ext.define("net.nan21.dnet.module.bp.md.ds.param.BusinessPartnerDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 

File: nan21.dnet.module.bp/nan21.dnet.module.bp.presenter/src/main/java/net/nan21/dnet/module/bp/md/ds/model/BusinessPartnerDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2010 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.bp.md.ds.model;

import java.util.Date;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.api.model.IModelWithClientId;
import net.nan21.dnet.core.api.model.IModelWithId;
import net.nan21.dnet.core.presenter.model.AbstractDsModel;

import net.nan21.dnet.module.bp.md.domain.entity.BusinessPartner;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = BusinessPartner.class, sort = { @SortField(field = BusinessPartnerDs.fNAME) })
public class BusinessPartnerDs extends AbstractDsModel<BusinessPartner>
        implements IModelWithId, IModelWithClientId {

    public static final String fNAME = "name";
    public static final String fCODE = "code";
    public static final String fACTIVE = "active";
    public static final String fNOTES = "notes";
    public static final String fID = "id";
    public static final String fCLIENTID = "clientId";
    public static final String fCREATEDAT = "createdAt";
    public static final String fMODIFIEDAT = "modifiedAt";
    public static final String fCREATEDBY = "createdBy";
    public static final String fMODIFIEDBY = "modifiedBy";
    public static final String fVERSION = "version";
    public static final String fTYPE = "type";
    public static final String fBUSINESSOBJECT = "businessObject";
    public static final String fCOUNTRYID = "countryId";
    public static final String fCOUNTRYCODE = "countryCode";
    public static final String fTAXPAYERNO = "taxPayerNo";
    public static final String fFIRSTNAME = "firstName";
    public static final String fLASTNAME = "lastName";
    public static final String fMIDDLENAME = "middleName";
    public static final String fGENDER = "gender";
    public static final String fIDENTITYCARDNO = "identityCardNo";
    public static final String fPASSPORTNO = "passportNo";
    public static final String fBIRTHDATE = "birthDate";
    public static final String fCOMPANYNAME = "companyName";
    public static final String fLEGALFORMID = "legalFormId";
    public static final String fLEGALFORM = "legalForm";
    public static final String fREGISTRATIONNO = "registrationNo";
    public static final String fREGISTRATIONDATE = "registrationDate";
    public static final String fCLASSNAME = "className";

    @DsField()
    private String name;

    @DsField()
    private String code;

    @DsField()
    private Boolean active;

    @DsField()
    private String notes;

    @DsField()
    private Long id;

    @DsField()
    private Long clientId;

    @DsField()
    private Date createdAt;

    @DsField()
    private Date modifiedAt;

    @DsField()
    private String createdBy;

    @DsField()
    private String modifiedBy;

    @DsField()
    private Long version;

    @DsField()
    private String type;

    @DsField(fetch = false)
    private String businessObject;

    @DsField(join = "left", path = "country.id")
    private Long countryId;

    @DsField(join = "left", path = "country.code")
    private String countryCode;

    @DsField()
    private String taxPayerNo;

    @DsField()
    private String firstName;

    @DsField()
    private String lastName;

    @DsField()
    private String middleName;

    @DsField()
    private String gender;

    @DsField()
    private String identityCardNo;

    @DsField()
    private String passportNo;

    @DsField()
    private Date birthDate;

    @DsField()
    private String companyName;

    @DsField(join = "left", path = "legalForm.id")
    private Long legalFormId;

    @DsField(join = "left", path = "legalForm.name")
    private String legalForm;

    @DsField()
    private String registrationNo;

    @DsField()
    private Date registrationDate;

    @DsField(fetch = false)
    private String className;

    public BusinessPartnerDs() {
        super();
    }

    public BusinessPartnerDs(BusinessPartner e) {
        super(e);
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return this.code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public Boolean getActive() {
        return this.active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }

    public String getNotes() {
        return this.notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Object id) {
        this.id = this._asLong_(id);

    }

    public Long getClientId() {
        return this.clientId;
    }

    public void setClientId(Long clientId) {
        this.clientId = clientId;
    }

    public Date getCreatedAt() {
        return this.createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getModifiedAt() {
        return this.modifiedAt;
    }

    public void setModifiedAt(Date modifiedAt) {
        this.modifiedAt = modifiedAt;
    }

    public String getCreatedBy() {
        return this.createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    public String getModifiedBy() {
        return this.modifiedBy;
    }

    public void setModifiedBy(String modifiedBy) {
        this.modifiedBy = modifiedBy;
    }

    public Long getVersion() {
        return this.version;
    }

    public void setVersion(Long version) {
        this.version = version;
    }

    public String getType() {
        return this.type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getBusinessObject() {
        return this.businessObject;
    }

    public void setBusinessObject(String businessObject) {
        this.businessObject = businessObject;
    }

    public Long getCountryId() {
        return this.countryId;
    }

    public void setCountryId(Long countryId) {
        this.countryId = countryId;
    }

    public String getCountryCode() {
        return this.countryCode;
    }

    public void setCountryCode(String countryCode) {
        this.countryCode = countryCode;
    }

    public String getTaxPayerNo() {
        return this.taxPayerNo;
    }

    public void setTaxPayerNo(String taxPayerNo) {
        this.taxPayerNo = taxPayerNo;
    }

    public String getFirstName() {
        return this.firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return this.lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getMiddleName() {
        return this.middleName;
    }

    public void setMiddleName(String middleName) {
        this.middleName = middleName;
    }

    public String getGender() {
        return this.gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getIdentityCardNo() {
        return this.identityCardNo;
    }

    public void setIdentityCardNo(String identityCardNo) {
        this.identityCardNo = identityCardNo;
    }

    public String getPassportNo() {
        return this.passportNo;
    }

    public void setPassportNo(String passportNo) {
        this.passportNo = passportNo;
    }

    public Date getBirthDate() {
        return this.birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public String getCompanyName() {
        return this.companyName;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    public Long getLegalFormId() {
        return this.legalFormId;
    }

    public void setLegalFormId(Long legalFormId) {
        this.legalFormId = legalFormId;
    }

    public String getLegalForm() {
        return this.legalForm;
    }

    public void setLegalForm(String legalForm) {
        this.legalForm = legalForm;
    }

    public String getRegistrationNo() {
        return this.registrationNo;
    }

    public void setRegistrationNo(String registrationNo) {
        this.registrationNo = registrationNo;
    }

    public Date getRegistrationDate() {
        return this.registrationDate;
    }

    public void setRegistrationDate(Date registrationDate) {
        this.registrationDate = registrationDate;
    }

    public String getClassName() {
        return this.className;
    }

    public void setClassName(String className) {
        this.className = className;
    }

}
